olp-cpp-sdk-core/include/olp/core/thread/TaskSchedulerFactory.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <olp/core/CoreApi.h>
+#include <olp/core/porting/make_unique.h>
+
+#include "ThreadPoolTaskScheduler.h"
+
+namespace olp {
+namespace thread {
+class TaskScheduler;
+}
+
+namespace thread {
+/**
+ * @brief Factory that creates TaskScheduler instances to be used used for all
+ * delayed operations.
+ */
+class CORE_API TaskSchedulerFactory {
+ public:
+  virtual ~TaskSchedulerFactory() = default;
+
+  /**
+   * @brief Creates an TaskScheduler instance used for all delayed operations.
+   *
+   * @return An instance of TaskScheduler.
+   */
+  static std::unique_ptr<TaskScheduler> Create() {
+    return std::make_unique<ThreadPoolTaskScheduler>();
```

Comment:
There is a very important constructor parameter that should pe fwd to the user:

```cpp
/**
   * @brief Constructor with default thread_count set to 1.
   * @param[in] thread_count Number of threads to be initialized in the thread
   * pool.
   */
  ThreadPoolTaskScheduler(size_t thread_count = 1u);
```

Reply:
For the default scheduler we should provide the most trivial initialization. Once user wants to setup a different one, non-default initialization should be used.
I will add the default parameter, but let's keep this point under discussion.